src/Npgsql/NpgsqlConnector.cs

```diff
@@ -494,10 +500,38 @@ internal async Task LoadDatabaseInfo(bool forceReload, NpgsqlTimeout timeout, bo
             // being set up (even if its empty)
             TypeMapper = new ConnectorTypeMapper(this);
 
-            if (forceReload || !NpgsqlDatabaseInfo.Cache.TryGetValue(ConnectionString, out var database))
-                NpgsqlDatabaseInfo.Cache[ConnectionString] = database = await NpgsqlDatabaseInfo.Load(Connection, timeout, async);
+            NpgsqlDatabaseInfo? database;
+
+            if (forceReload)
+            {
+                NpgsqlDatabaseInfo.Cache[ConnectionString] = database = await NpgsqlDatabaseInfo.Load(Connection,
+                    timeout, async);
+            }
```

Comment:
`forceReload` should be rare, so I think it isn't worth to have a separate info loading outside of the semaphore.

Reply:
I completely agree! We can simplify this and use the same path.